pkg/controller/blockdeviceclaim/blockdeviceclaim_controller.go

```diff
@@ -270,22 +282,27 @@ func (r *ReconcileBlockDeviceClaim) releaseClaimedBlockDevice(
 	}
 
 	// Check if same deviceclaim holding the ObjRef
+	var claimedBd *apis.BlockDevice
 	for _, item := range bdList.Items {
-		if !r.isDeviceRequestedByThisDeviceClaim(instance, item) {
-			continue
-		}
-
 		// Found a blockdevice ObjRef with BlockDeviceClaim, Clear
 		// ObjRef and mark blockdevice released in etcd
-		dvr := item.DeepCopy()
-		dvr.Spec.ClaimRef = nil
-		dvr.Status.ClaimState = apis.BlockDeviceReleased
-		err := r.client.Update(context.TODO(), dvr)
-		if err != nil {
-			klog.Errorf("Error updating ClaimRef of %s: %v", dvr.Name, err)
-			return err
+		if r.isDeviceRequestedByThisDeviceClaim(instance, item) {
+			claimedBd = &item
+			break
 		}
+		claimedBd = &item
```

Comment:
We are already getting claimed BD inside the `if` condition. This can be removed.